### PR TITLE
DAOS-9990: Fix various coverity issues

### PR DESCRIPTION
**Pull request by dpquigl**
Coverity identified some dead code and leaked resources. This patch fixes
resource leaks in the compression and ring_map code and removes an unchecked
return code assignment.

Signed-off-by: David Quigley <david.quigley@intel.com>

### CODE REVIEW

File: src/placement/ring_map.c [hunk 1]
```diff
@@ -1253,8 +1253,10 @@ ring_obj_find_reint(struct pl_map *map, struct daos_obj_md *md,
 			return rc;
 		rc = pl_obj_layout_alloc(rop.rop_grp_size, rop.rop_grp_nr,
 				&reint_layout);
-		if (rc)
+		if (rc) {
+			D_FREE(layout);
```
Comment (jolivier23) [inline src/placement/ring_map.c:1257]: I just reviewed this same patch against 2.0.  See the comments there.   The process for this is to get patch landed on master before cherry picking it to 2.0.  I don't see much reason to diverge from this here.

Anyway, the comment I made here on the other is that this should be free'd inside of pl_obj_layout_alloc if it returns an error.  If it isn't, it's a bug in that routine.

**Review by jolivier23 — CHANGES_REQUESTED** [review]
Setting to request changes as with other patch.

File: src/placement/ring_map.c (hunk 1 above)
Comment (dpquigl) [reply src/placement/ring_map.c:1257]: The issue is layout is successful but the second pl_obj_layout_alloc fails leaving layout leaked.

Comment (jolivier23) [reply src/placement/ring_map.c:1257]: ah, I see. I missed that it was for layout.

**Review by jolivier23 — APPROVED** [review]

**Review by Nasf-Fan — APPROVED** [review]

**Pull request merged**